packages/kbn-capture-oas-snapshot-cli/src/run_capture_oas_snapshot_cli.ts

```diff
@@ -29,100 +21,65 @@ export const sortAndPrettyPrint = (object: object) => {
-      const { msg$, proc } = startTSWorker<Result>({
+    if (traditional) {
+      log.info('Capturing OAS for traditional Kibana...');
+      await captureOasSnapshot({
         log,
-        src: require.resolve('./kibana_worker'),
+        buildFlavour: 'traditional',
+        outputFile: path.resolve(OAS_OUTPUT_DIR, 'bundle.json'),
+        filters: { pathStartsWith, excludePathsMatching },
+        update,
       });
-      await Rx.firstValueFrom(
-        msg$.pipe(
-          Rx.map((msg) => {
-            if (msg !== 'ready')
-              throw new Error(`received unexpected message from worker (expected "ready"): ${msg}`);
-          })
-        )
-      );
-      addCleanupTask(() => proc.kill('SIGILL'));
-    });
+      log.success('Captured OAS for traditional Kibana.');
+    }
 
-    try {
-      const qs = encode({
-        access: 'public',
-        version: '2023-10-31', // hard coded for now, we can make this configurable later
-        pathStartsWith,
-        excludePathsMatching,
-      });
-      const url = `http://localhost:${port}/api/oas?${qs}`;
-      log.info(`Fetching OAS at ${url}...`);
-      const result = await fetch(url, {
-        headers: {
-          'kbn-xsrf': 'kbn-oas-snapshot',
-          authorization: `Basic ${Buffer.from('elastic:changeme').toString('base64')}`,
-        },
+    if (serverless) {
+      log.info('Capturing OAS for serverless Kibana...');
+      await captureOasSnapshot({
+        log,
+        buildFlavour: 'serverless',
+        outputFile: path.resolve(OAS_OUTPUT_DIR, 'bundle.serverless.json'),
+        filters: { pathStartsWith, excludePathsMatching },
+        update,
       });
-      if (result.status !== 200) {
-        log.error(`Failed to fetch OAS: ${JSON.stringify(result, null, 2)}`);
-        throw new Error(`Failed to fetch OAS: ${result.status}`);
-      }
-      const currentOas = await result.json();
-      log.info(`Recieved OAS, writing to ${OAS_FILE_PATH}...`);
-      if (update) {
-        await fs.writeFile(OAS_FILE_PATH, sortAndPrettyPrint(currentOas));
-        const { size: sizeBytes } = await fs.stat(OAS_FILE_PATH);
-        log.success(`OAS written to ${OAS_FILE_PATH}. File size ~${twoDeci(sizeBytes / MB)} MB.`);
-      } else {
-        log.success(
-          `OAS recieved, not writing to file. Got OAS for ${
-            Object.keys(currentOas.paths).length
-          } paths.`
-        );
-      }
-    } catch (err) {
-      log.error(`Failed to capture OAS: ${JSON.stringify(err, null, 2)}`);
-      throw err;
+      log.success('Captured OAS for serverless Kibana.');
     }
   },
   {
     description: `
       Get the current OAS from Kibana's /api/oas API
     `,
     flags: {
-      boolean: ['update'],
-      string: ['include-path', 'exclude-path'],
+      boolean: ['update', 'serverless', 'traditional'],
+      string: ['include-path', 'exclude-path', 'build-flavor'],
```

Comment:
What is this `build-flavor` flag? was it the first approach you took before switching to boolean flags? it's not used in the PR, right?

Reply:
yes correct, this should have been removed in 782381a3dd8